converter/src/main/java/gov/cms/qpp/conversion/model/TemplateId.java

```diff
@@ -128,16 +131,19 @@ public String getTemplateId(Context context) {
 	 */
 	public static TemplateId getTemplateId(String root, String extension, Context context) {
 		Map<String, TemplateId> extensionsToTemplateId = ROOT_AND_TO_TEMPLATE_ID.get(root);
+		TemplateId retrieved = null;
 
 		if (extensionsToTemplateId == null) {
-			return TemplateId.DEFAULT;
-		}
-
-		if (context.isHistorical()) {
-			return extensionsToTemplateId.getOrDefault(null, TemplateId.DEFAULT);
+			retrieved = TemplateId.DEFAULT;
+		} else if (context.isHistorical()) {
+			retrieved = extensionsToTemplateId.getOrDefault(null, TemplateId.DEFAULT);
+		} else {
+			retrieved = EnvironmentHelper.isPresent(Extension.STRICT_EXTENSION) ?
```

Comment:
When doing loose extension checking, I'm leaning towards keeping the strict extension checking here for the clinical document.  Right now, this logic would return a clinical document templateId if only the root was correct.  If we made this strict for the clinical document, whenever we get a `CLINICAL_DOCUMENT` from  this method, we wouldn't have to do a subsequent check to _really_ ensure it is a real one.

I can see benefits going the way you went too, though.  Thoughts?

Reply:
We do need to keep the strict extension checking for the clinical document template id.  That's what was discussed previously.

Reply:
@halprin so you're suggesting `TemplateId.CLINICAL_DOCUMENT.root.equals(root) || EnvironmentHelper.isPresent(Extension.STRICT_EXTENSION)` as a condition for the ternary expression?

Reply:
Yes.